src/migrateDashboard.ts

```diff
@@ -61,6 +61,61 @@ function addWidgetErrorToReport(
   };
 }
 
+/*
+    ensure that unexpected elements in the dashboard and page filter objects are removed, this will prevent avoidable migration failures
+    Also keep track of what dashboards and pages have had this treatment
+    It is expected that this treatment will have no negative impact because the unexpected elements would have been ignored by AUI4 anyway
+    but keep track for customer's peace of mind
+  */
+const sanitiseFilters = (
+  filters: any[],
+  errorReport: DashboardErrorReport,
+  {
+    doesReportIncludeStacks,
+    pageKey,
+    pageName,
+  }: {
+    doesReportIncludeStacks?: boolean;
+    pageKey: string;
+    pageName: string;
+  },
+) => {
+  const error: {
+    message: string;
+    stack?: string[];
+  } = {
+    message: "Incompatible filters were removed",
+  };
+  const rejectedFilters: string[] = [];
+  const acceptedFilters = filters.filter((element) => {
+    if (typeof element === "string") {
+      return true;
+    } else {
+      rejectedFilters.push(
+        typeof element === "object" ? JSON.stringify(element) : element,
+      );
+      return false;
+    }
+  });
+  if (rejectedFilters.length > 0) {
+    const sectionKey = pageKey;
+    if (!errorReport.pages[sectionKey]) {
+      errorReport.pages[sectionKey] = {
+        pageName,
+        widgets: {},
+      };
+    }
+
+    if (doesReportIncludeStacks) error.stack = rejectedFilters;
+
+    errorReport.pages[sectionKey].widgets[`All widgets in ${pageName}`] = {
+      widgetName: "All widgets",
+      error,
+    };
```

Comment:
This is a bit of a workaround.
I would prefer to introduce an `error` at the root of `DashboardErrorReport` and at the root of each page if necessary, in a preliminary PR.